Automattic/wp-calypso
# Magic Flow Updates for WooPayments

Related to #95664

## Proposed Changes

These changes are related to addressing some confusion caused when merchants are bounced from their store to Calypso as part of the WooPayments onboarding. For more info, please see #95664.

* Hide the psuedo-generated username when a user comes through the passwordless flow to connect to WooPayments (only affects Woo-DNA onboardings).
* Hide the Jetpack branding when a user accepts a magic login link and they are in the process of a WooPayments onboarding.

## Why are these changes being made?


* The psuedo-generated wordpress.com username and Jetpack branding can cause confusion for a WooPayments merchant being onboarded. A jetpack connection is necessary for the user to use WooPayments, but streamlining this flow and using consistent branding helps reduce friction for merchants.

## Testing Instructions



Setup (h/t @chihsuan):

> You can use a reverse proxy (nginx) in front of the calypso server PCYsg-5YE-p2 or use [woocommerce-start-dev-env](https://github.com/woocommerce/woocommerce-start-dev-env), which our team created to help with local development (you can skip wccommerce.com part)

0. Using a new store that doesn't already have a Jetpack account, onboard a new account onto WooPayments by installing the WooPayments plugin and going to `Payments` in WP Admin.
1. The first step will involve connecting your Jetpack account. Use an incognito browser where you aren't logged in to your Wordpress.com account, and enter an email address that doesn't already have an account registered (you can use `+test`, for example `your.name+test@automattic.com`
2. Go to your email and click the magic link that gets sent.
3. Verify that when you are taken back to the authorize page, you do **not** see the psuedorandom username generated - you should see `Connecting your account` / `Account connected successfully`.
4. Before being taken back to the WooPayments onboarding, you will get redirected to a page that says `Email confirmed!` This page should _not_ show a Jetpack logo in the middle of the screen.
5. Verify the Jetpack connection was successfully established.

## Pre-merge Checklist





## Screenshots (for translators)
![Screenshot 2024-10-29 at 13 29 20](https://github.com/user-attachments/assets/b378d632-a20b-47a3-9e14-aecac941ee42)
![Screenshot 2024-10-29 at 13 29 30](https://github.com/user-attachments/assets/8937d3bf-bd08-45bd-af52-a79dbde5a156)
![Screenshot 2024-10-29 at 13 29 42](https://github.com/user-attachments/assets/66cabc86-fb51-485c-9cbb-a39cb8c0744b)
